### PR TITLE
Add support i1 and bv1 for languages that use these types

### PR DESCRIPTION
**Pull request by keram88**
Adds support for the 1-bit integer type for regular integers, and sign extension/truncation for 1-bit bit-vectors. The Rust compiler emits i1 and bv1 (in bv mode) for the boolean type. Other languages may require this feature as well.

### CODE REVIEW

**Comment by zvonimir** [comment]
First of all, please write better commit messages in the future. Read our contribution guidelines on commit messages. This time around you can leave it as is.
Second, would it be possible to add a C regression that loads/stores i1? I suspect that is not easy and I don't see how to do it. Maybe try bit-fields?
Otherwise, this looks good.

**Comment by keram88** [comment]
A Clang-specific regression is to use the **__builtin_add_overflow** intrinsic, or a related checked-arithmetic intrinsic. This produces code similar to
```
  %8 = call { i32, i1 } @llvm.uadd.with.overflow.i32(i32 %6, i32 %7)
  %9 = extractvalue { i32, i1 } %8, 1
  %10 = extractvalue { i32, i1 } %8, 0
```

"extractvalue { i32, i1 } %8, 1" is transformed into "load i1 ..." in the StructReturnToPointer pass. However, this requires simultaneous support for the llvm overflow intrinsics.

**Comment by shaobo-he** [comment]
LGTM

File: share/smack/lib/smack.c [hunk 1]
```diff
@@ -1455,6 +1456,7 @@ void __SMACK_decls(void) {
   D("function {:inline} $store.bv24(M: [ref] bv24, p: ref, v: bv24) returns ([ref] bv24) { M[p := v] }");
   D("function {:inline} $store.bv16(M: [ref] bv16, p: ref, v: bv16) returns ([ref] bv16) { M[p := v] }");
   D("function {:inline} $store.bv8(M: [ref] bv8, p: ref, v: bv8) returns ([ref] bv8) { M[p := v] }");
+  D("function {:inline} $store.bv8(M: [ref] bv1, p: ref, v: bv1) returns ([ref] bv1) { M[p := v] }");
```
Comment (shaobo-he) [inline share/smack/lib/smack.c]: A typo. Should be `$store.bv1`.

**Pull request merged**